internal/pkg/manifest/validate.go

```diff
@@ -734,3 +812,108 @@ func (d *DeadLetterQueue) Validate() error {
 func (*OverrideRule) Validate() error {
 	return nil
 }
+
+type validateDependenciesOpts struct {
+	mainContainerName string
+	sidecarConfig     map[string]*SidecarConfig
+	imageConfig       *Image
+}
+
+func validateDependencies(opts validateDependenciesOpts) error {
+	if err := validateImageDependsOnStatus(opts); err != nil {
+		return err
+	}
+	if err := validateSidecarsDependsOnStatus(opts); err != nil {
+		return err
+	}
+	return validateNoCircularDependencies(opts)
+}
+
+func validateImageDependsOnStatus(opts validateDependenciesOpts) error {
+	if opts.imageConfig == nil || opts.imageConfig.DependsOn == nil {
+		return nil
+	}
+	for name, status := range opts.imageConfig.DependsOn {
+		if !isEssentialContainer(name, opts) {
+			continue
+		}
+		if err := isValidEssentialStatus(name, strings.ToUpper(status)); err != nil {
+			return fmt.Errorf("validate %s container dependencies status: %w", opts.mainContainerName, err)
```

Comment:
What do you think of this error message:
```suggestion
			return fmt.Errorf("validate %s container "depends_on": %w", opts.mainContainerName, err)
```
and samesies in `validateSidecarsDependsOnStatus`

Reply:
hmm the reason is because we have https://github.com/aws/copilot-cli/blob/807ae3f5fc370397c3220cd53464db087176f26e/internal/pkg/manifest/validate.go#L270 and also we are not validating only in either `Image` or `Sidecar` level. This validation serves more like a general dependency check as opposed to just one field.